script.sql

```diff
@@ -0,0 +1,162 @@
+
+
+-- tables
+-- Table: app
+CREATE TABLE app (
+    id int NOT NULL AUTO_INCREMENT,
+    name varchar(25) NOT NULL,
+    description text NOT NULL,
+    weight int NOT NULL,
+    create_date date NOT NULL,
+    is_free bool NOT NULL,
+    price decimal(8,2) NULL,
+    has_advert bool NOT NULL,
+    category_id int NOT NULL,
+    CONSTRAINT app_pk PRIMARY KEY (id)
+);
+
+-- Table: app_category
+CREATE TABLE app_category (
+    id int NOT NULL AUTO_INCREMENT,
+    name varchar(30) NOT NULL,
+    description text NULL,
+    CONSTRAINT app_category_pk PRIMARY KEY (id)
+);
+
+-- Table: app_creating
+CREATE TABLE app_creating (
+    app_id int NOT NULL,
+    creater_id int NOT NULL,
+    CONSTRAINT app_creating_pk PRIMARY KEY (app_id,creater_id)
+);
+
+-- Table: card
+CREATE TABLE card (
+    number int NOT NULL,
+    bank_name varchar(25) NOT NULL,
+    user_id int NOT NULL,
+    CONSTRAINT card_pk PRIMARY KEY (number)
+);
+
+-- Table: category
+CREATE TABLE category (
+    id int NOT NULL AUTO_INCREMENT,
+    audience_type enum ('child', 'teen', 'mature', 'all') NOT NULL,
+    app_category_id int NOT NULL,
+    CONSTRAINT category_pk PRIMARY KEY (id)
+);
+
+-- Table: creater
+CREATE TABLE creater (
+    id int NOT NULL AUTO_INCREMENT,
+    full_name varchar(60) NOT NULL,
+    work_branch varchar(50) NOT NULL,
+    email varchar(70) NOT NULL,
+    created_app int NULL,
+    CONSTRAINT creater_pk PRIMARY KEY (id)
+);
+
+-- Table: device_type
+CREATE TABLE device_type (
+    device enum('mobile', 'tablet', 'chromebook', 'tv') NOT NULL,
+    app_id int NOT NULL,
+    CONSTRAINT device_type_pk PRIMARY KEY (device)
+);
+
+-- Table: download
+CREATE TABLE download (
+    id int NOT NULL AUTO_INCREMENT,
+    amount int NOT NULL,
+    total_price decimal(8,2) NOT NULL,
+    user_id int NOT NULL,
+    CONSTRAINT download_pk PRIMARY KEY (id)
+);
+
+-- Table: download_info
+CREATE TABLE download_info (
+    downloaded_time time NOT NULL,
+    app_id int NOT NULL,
+    download_id int NOT NULL,
+    CONSTRAINT download_info_pk PRIMARY KEY (app_id,download_id)
+);
+
+-- Table: email_preferences
+CREATE TABLE email_preferences (
+    email varchar(100) NOT NULL AUTO_INCREMENT,
+    keep_with_up_to_date_news bool NOT NULL,
+    receive_reply_notification  bool NOT NULL,
+    user_id int NOT NULL,
+    CONSTRAINT email_preferences_pk PRIMARY KEY (email)
+);
+
+-- Table: feedback
+CREATE TABLE feedback (
+    id int NOT NULL AUTO_INCREMENT,
+    description text NULL,
+    created_date date NOT NULL,
+    rate int NOT NULL,
+    user_id int NOT NULL,
+    app_id int NOT NULL,
+    CONSTRAINT feedback_pk PRIMARY KEY (id)
+);
+
+-- Table: user
+CREATE TABLE user (
+    id int NOT NULL AUTO_INCREMENT,
+    password varchar(50) NOT NULL,
```

Comment:
Just so you know, typically you don't store the password in your DB, you store a salt (salt is an encrypted version of password that takes much longer to crack, and when a user logs in, you encrypt the password they provided and compare it with the salt stored) or use an external auth system (e.g. Firebase Auth, AWS Cognito etc.).

This is not required for this task, but just so you know for your future work experience.